modules/planning/lattice/trajectory_generation/trajectory_evaluator.cc

```diff
@@ -82,12 +82,13 @@ TrajectoryEvaluator::TrajectoryEvaluator(
       */
       if (!FLAGS_enable_auto_tuning) {
         double cost = Evaluate(planning_target, lon_trajectory, lat_trajectory);
-        cost_queue_.push(PairCost({lon_trajectory, lat_trajectory}, cost));
+        // Use more efficient "emplace" member fucntion. 
+        cost_queue_.emplace(PairCost({lon_trajectory, lat_trajectory}, cost));
       } else {
         std::vector<double> cost_components;
         double cost = Evaluate(planning_target, lon_trajectory, lat_trajectory,
                                &cost_components);
-        cost_queue_with_components_.push(PairCostWithComponents(
+        cost_queue_with_components_.emplace(PairCostWithComponents(
```

Comment:
this is not necessary. push or emplace here have no difference.
It is generally preferred to keeping using push if there is no performance improvement.

Reply:
Please use cost_queue_with_components_.emplace(lon_trajectory, lat_trajectory).

The way you wrote (emplace(PairCostWithComponents{xxx, xxx}) requires a construction of temporary instance and invokes the copy constructor, which is identical to what you will get with a push().

Reply:
Great, thank you for pointing out my problems. I will correct them right away.

Reply:
I have modified the code, please check again. You are awesome!